geniza/corpus/models.py

```diff
@@ -525,6 +525,23 @@ def editions(self):
             "content", "source"
         )
 
+    def digital_editions(self):
+        """All footnotes for this document where the document relation includes
+        edition AND the footnote has content."""
+        editions_with_content = (
+            self.footnotes.filter(doc_relation__contains=Footnote.EDITION)
+            .filter(content__isnull=False)
+            .order_by("content", "source")
+        )
+        display_strings = []
+        editions_with_unique_display = []
+        for f in editions_with_content:
+            if f.display() not in display_strings:
+                # Prevent redundant display of identical brief citations on detail page
+                editions_with_unique_display += [f.pk]
+                display_strings += [f.display()]
+        return editions_with_content.filter(pk__in=editions_with_unique_display)
```

Comment:
@rlskoeser before I request your full review on this, I had a quick question. Does it make sense for me to leave this filtering of "duplicate" entries in the brief citations in this `Document` model function, or would it make more sense for this logic to be on the document detail view in `views.py` instead? It feels a bit strange as I have it here, since you might expect `digital_editions()` to return all digital editions, regardless of whether or not their brief citations would appear identical once rendered as strings.

Reply:
To see an example of why I did this, you can take a look at this document on the existing test site: https://test-geniza.cdh.princeton.edu/en/documents/2142/

Reply:
I agree, it feels strange here. I think the duplicates are likely a result of some footnote cleanup that needs to be done for the merged records, so it should hopefully be a temporary problem. I'd rather see the deduplication kept to the template — what about using `ifchanged` in the template to omit redundant citations?

Reply:
Thanks, didn't know about that template tag! Running into one problem with this, which is counting editors when there might be duplicates. For example, https://test-geniza.cdh.princeton.edu/en/documents/2493 will read "Editors" while displaying only one brief citation. Maybe just a quick view method like `editor_count` or something?

Reply:
right. editor(s) should be based on distinct number of edition authors anyway, right? so generating a distinct list that gives you access to count makes sense to me